simpleviewinc/keg-cli
# Keg switch to args-parse package

## Context

* Extract argument parsing to an external library [args-parse](https://github.com/lancetipton/argsParse)
* Allows using the external library in other repos [keg-regulator](https://github.com/simpleviewinc/keg-regulator)

## Goal

* Ensure the cli still works with the external library
* There should be no noticeable difference from these changes
* Remove unused code


## Updates

* `package.json`
  * Added arg-parse lib from npm
* Removed unused code due to updates
  * `src/constants/constants.js`
  * `src/utils/task/getParams.js`
  * `src/utils/task/index.js`
  * `src/utils/task/taskOptions.js`
* `src/utils/helpers/__tests__/optionsHasArg.js`
  * Added some tests due to the changes
* Updated to use arg-parse lib
  * `src/utils/helpers/parseArgs.js`
  * `src/utils/task/executeTask.js`
  * `src/utils/task/findTask.js`
  * `src/utils/task/runInternalTask.js`
  * `src/utils/task/optionsAsk.js`


## Testing

* Pull branch, and run `yarn test`
  * Ensure all tests pass
* Run cli task `keg test --foo fax --fax boo`
  * Enter a value for `What sound casper makes`
  * Ensure the output look like this
	```js
      ---------- Keg-CLI Validate ----------
      {
        foo: 'fax',
        fax: 'boo',
        boo: 'OoOoOoO',
        do: undefined,
        env: 'development'
      }
    ```
* Run any other keg task and ensure it runs as expected
